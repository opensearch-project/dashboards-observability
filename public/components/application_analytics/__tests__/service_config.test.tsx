/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { waitFor } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import DSLService from 'public/services/requests/dsl';
import React from 'react';
import { coreStartMock } from '../../../../test/__mocks__/coreMocks';
import { ServiceConfig } from '../components/config_components/service_config';

describe('Service Config component', () => {
  configure({ adapter: new Adapter() });

  it('renders empty service config', async () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setSelectedServices = jest.fn();
    const setNameWithStorage = jest.fn();
    const setDescriptionWithStorage = jest.fn();
    const setQueryWithStorage = jest.fn();
    const setFiltersWithStorage = jest.fn();
    const setAppConfigs = jest.fn();
    const setStartTimeWithStorage = jest.fn();
    const setEndTimeWithStorage = jest.fn();
    const dslService = ({
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn(),
    } as unknown) as DSLService;
    const wrapper = mount(
      <ServiceConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query=""
        setQuery={setQuery}
        filters={[]}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        mode="data_prepper"
        dslService={dslService}
        selectedServices={[]}
        setSelectedServices={setSelectedServices}
        name=""
        description=""
        setNameWithStorage={setNameWithStorage}
        setDescriptionWithStorage={setDescriptionWithStorage}
        setQueryWithStorage={setQueryWithStorage}
        setFiltersWithStorage={setFiltersWithStorage}
        appConfigs={[]}
        setAppConfigs={setAppConfigs}
        setStartTimeWithStorage={setStartTimeWithStorage}
        setEndTimeWithStorage={setEndTimeWithStorage}
      />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });

  it('renders with one service selected', async () => {
    const core = coreStartMock;
    const setQuery = jest.fn();
    const setFilters = jest.fn();
    const setStartTime = jest.fn();
    const setEndTime = jest.fn();
    const setSelectedServices = jest.fn();
    const setNameWithStorage = jest.fn();
    const setDescriptionWithStorage = jest.fn();
    const setQueryWithStorage = jest.fn();
    const setFiltersWithStorage = jest.fn();
    const setAppConfigs = jest.fn();
    const setStartTimeWithStorage = jest.fn();
    const setEndTimeWithStorage = jest.fn();
    const dslService = ({
      http: jest.fn(),
      fetch: jest.fn(),
      fetchIndices: jest.fn(),
      fetchFields: jest.fn(),
    } as unknown) as DSLService;
    const serviceFilter = [
      {
        field: 'serviceName',
        operator: 'is',
        value: 'User',
        inverted: false,
        disabled: false,
      },
    ];
    const wrapper = mount(
      <ServiceConfig
        http={core.http}
        chrome={core.chrome}
        parentBreadcrumb={{ text: 'test', href: 'test#/' }}
        query=""
        setQuery={setQuery}
        filters={serviceFilter}
        setFilters={setFilters}
        startTime="now-5m"
        setStartTime={setStartTime}
        endTime="now"
        setEndTime={setEndTime}
        mode="data_prepper"
        dslService={dslService}
        selectedServices={[]}
        setSelectedServices={setSelectedServices}
        name=""
        description=""
        setNameWithStorage={setNameWithStorage}
        setDescriptionWithStorage={setDescriptionWithStorage}
        setQueryWithStorage={setQueryWithStorage}
        setFiltersWithStorage={setFiltersWithStorage}
        appConfigs={[]}
        setAppConfigs={setAppConfigs}
        setStartTimeWithStorage={setStartTimeWithStorage}
        setEndTimeWithStorage={setEndTimeWithStorage}
      />
    );

    wrapper.update();
    await waitFor(() => {
      expect(
        toJson(wrapper, {
          mode: 'deep',
        })
      ).toMatchSnapshot();
    });
  });
});
