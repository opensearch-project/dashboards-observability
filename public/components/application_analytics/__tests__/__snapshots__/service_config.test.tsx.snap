// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service Config component renders empty service config 1`] = `
<div>
  <div
    className="euiAccordion"
    data-test-subj="servicesEntitiesAccordion"
    onToggle={[Function]}
  >
    <div
      className="euiAccordion__triggerWrapper"
    >
      <button
        aria-controls="servicesEntities"
        aria-expanded={false}
        className="euiAccordion__button"
        id="random_html_id"
        onClick={[Function]}
        type="button"
      >
        <span
          className="euiAccordion__iconWrapper"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
        <span
          className="euiIEFlexWrapFix"
        >
          <div
            className="euiText euiText--small"
          >
            <h3>
              Services & entities
               
              <span
                className="euiBadge euiBadge--iconLeft"
                data-test-subj="servicesEntitiesCountBadge"
                style={
                  Object {
                    "backgroundColor": "#d3dae6",
                    "color": "#000",
                  }
                }
                title="0"
              >
                <span
                  className="euiBadge__content"
                >
                  0
                </span>
              </span>
            </h3>
          </div>
          <div
            className="euiSpacer euiSpacer--s"
          />
          <div
            className="euiText euiText--small"
          >
            <div
              className="euiTextColor euiTextColor--subdued"
            >
              Select services & entities to include in this application
            </div>
          </div>
        </span>
      </button>
      <div
        className="euiAccordion__optionalAction"
      >
        <button
          className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
          data-test-subj="clearServicesEntitiesButton"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "minWidth": undefined,
            }
          }
          type="button"
        >
          <span
            className="euiButtonContent euiButton__content"
          >
            <span
              className="euiButton__text"
            >
              Clear all
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      aria-labelledby="random_html_id"
      className="euiAccordion__childWrapper"
      id="servicesEntities"
      role="region"
      tabIndex={-1}
    >
      <div>
        <div
          className="euiAccordion__padding--l"
        >
          <div
            className="euiFormRow"
            id="random_html_id-row"
          >
            <div
              className="euiFormRow__labelWrapper"
            >
              <label
                className="euiFormLabel euiFormRow__label"
                htmlFor="random_html_id"
              >
                Services & entities
              </label>
            </div>
            <div
              className="euiFormRow__fieldWrapper"
            >
              <div
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-label="Select services and entities"
                className="euiComboBox"
                data-test-subj="servicesEntitiesComboBox"
                onKeyDown={[Function]}
                role="combobox"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <div
                      className="euiComboBox__inputWrap"
                      data-test-subj="comboBoxInput"
                      onClick={[Function]}
                      tabIndex={-1}
                    >
                      <p
                        className="euiComboBoxPlaceholder"
                      >
                        Select services and entities
                      </p>
                      <div
                        className="euiComboBox__input"
                        style={
                          Object {
                            "display": "inline-block",
                            "fontSize": 14,
                          }
                        }
                      >
                        <input
                          aria-controls=""
                          data-test-subj="comboBoxSearchInput"
                          id="random_html_id"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="textbox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "2px",
                            }
                          }
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        />
                      </div>
                    </div>
                    <div
                      className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                    >
                      <button
                        aria-label="Open list of options"
                        className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                        data-test-subj="comboBoxToggleListButton"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          height={16}
                          role="img"
                          style={null}
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="euiSpacer euiSpacer--l"
          />
          <div
            className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
          >
            <div
              className="euiText euiText--medium"
            >
              <span
                className="panel-title"
              >
                Service map
              </span>
            </div>
            <div
              className="euiSpacer euiSpacer--m"
            />
            <fieldset
              className="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
              disabled={false}
            >
              <legend
                className="euiScreenReaderOnly"
              />
              <div
                className="euiButtonGroup__buttons"
              >
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Duration"
                      title="Duration"
                    >
                      <input
                        checked={true}
                        className="euiScreenReaderOnly"
                        data-test-subj="latency"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Duration
                    </span>
                  </span>
                </label>
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Errors"
                      title="Errors"
                    >
                      <input
                        checked={false}
                        className="euiScreenReaderOnly"
                        data-test-subj="error_rate"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Errors
                    </span>
                  </span>
                </label>
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Request Rate"
                      title="Request Rate"
                    >
                      <input
                        checked={false}
                        className="euiScreenReaderOnly"
                        data-test-subj="throughput"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Request Rate
                    </span>
                  </span>
                </label>
              </div>
            </fieldset>
            <hr
              className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginMedium"
            />
            <div
              className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  className="euiText euiText--medium"
                >
                  Focus on
                </div>
              </div>
              <div
                className="euiFlexItem"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <input
                      className="euiFieldSearch"
                      onChange={[Function]}
                      onKeyUp={[Function]}
                      placeholder="Service name"
                      type="search"
                      value=""
                    />
                    <div
                      className="euiFormControlLayoutIcons"
                    >
                      <span
                        className="euiFormControlLayoutCustomIcon"
                      >
                        <svg
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          height={16}
                          role="img"
                          style={null}
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="euiSpacer euiSpacer--l"
            />
            <div
              style={
                Object {
                  "minHeight": 434,
                }
              }
            >
              Array [
                <div
                  className="euiSpacer euiSpacer--s"
                />,
                <div
                  className="euiEmptyPrompt"
                >
                  <h2
                    className="euiTitle euiTitle--medium"
                  >
                    No matches
                  </h2>
                  <span
                    className="euiTextColor euiTextColor--subdued"
                  >
                    <div
                      className="euiSpacer euiSpacer--m"
                    />
                    <div
                      className="euiText euiText--medium"
                    >
                      <div
                        className="euiText euiText--medium"
                      >
                        No data matches the selected filter. Clear the filter and/or increase the time range to see more results.
                      </div>
                    </div>
                  </span>
                </div>,
                <div
                  className="euiSpacer euiSpacer--s"
                />,
              ]
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Service Config component renders with one service selected 1`] = `
<div>
  <div
    className="euiAccordion"
    data-test-subj="servicesEntitiesAccordion"
    onToggle={[Function]}
  >
    <div
      className="euiAccordion__triggerWrapper"
    >
      <button
        aria-controls="servicesEntities"
        aria-expanded={false}
        className="euiAccordion__button"
        id="random_html_id"
        onClick={[Function]}
        type="button"
      >
        <span
          className="euiAccordion__iconWrapper"
        >
          <svg
            aria-hidden={true}
            className="euiIcon euiIcon--medium euiAccordion__icon"
            focusable="false"
            height={16}
            role="img"
            style={null}
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.157 13.069 4.611-4.685a.546.546 0 0 0 0-.768L5.158 2.93a.552.552 0 0 1 0-.771.53.53 0 0 1 .759 0l4.61 4.684c.631.641.63 1.672 0 2.312l-4.61 4.684a.53.53 0 0 1-.76 0 .552.552 0 0 1 0-.771Z"
              fillRule="nonzero"
            />
          </svg>
        </span>
        <span
          className="euiIEFlexWrapFix"
        >
          <div
            className="euiText euiText--small"
          >
            <h3>
              Services & entities
               
              <span
                className="euiBadge euiBadge--iconLeft"
                data-test-subj="servicesEntitiesCountBadge"
                style={
                  Object {
                    "backgroundColor": "#d3dae6",
                    "color": "#000",
                  }
                }
                title="0"
              >
                <span
                  className="euiBadge__content"
                >
                  0
                </span>
              </span>
            </h3>
          </div>
          <div
            className="euiSpacer euiSpacer--s"
          />
          <div
            className="euiText euiText--small"
          >
            <div
              className="euiTextColor euiTextColor--subdued"
            >
              Select services & entities to include in this application
            </div>
          </div>
        </span>
      </button>
      <div
        className="euiAccordion__optionalAction"
      >
        <button
          className="euiButton euiButton--primary euiButton--small euiButton-isDisabled"
          data-test-subj="clearServicesEntitiesButton"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "minWidth": undefined,
            }
          }
          type="button"
        >
          <span
            className="euiButtonContent euiButton__content"
          >
            <span
              className="euiButton__text"
            >
              Clear all
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      aria-labelledby="random_html_id"
      className="euiAccordion__childWrapper"
      id="servicesEntities"
      role="region"
      tabIndex={-1}
    >
      <div>
        <div
          className="euiAccordion__padding--l"
        >
          <div
            className="euiFormRow"
            id="random_html_id-row"
          >
            <div
              className="euiFormRow__labelWrapper"
            >
              <label
                className="euiFormLabel euiFormRow__label"
                htmlFor="random_html_id"
              >
                Services & entities
              </label>
            </div>
            <div
              className="euiFormRow__fieldWrapper"
            >
              <div
                aria-expanded={false}
                aria-haspopup="listbox"
                aria-label="Select services and entities"
                className="euiComboBox"
                data-test-subj="servicesEntitiesComboBox"
                onKeyDown={[Function]}
                role="combobox"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <div
                      className="euiComboBox__inputWrap"
                      data-test-subj="comboBoxInput"
                      onClick={[Function]}
                      tabIndex={-1}
                    >
                      <p
                        className="euiComboBoxPlaceholder"
                      >
                        Select services and entities
                      </p>
                      <div
                        className="euiComboBox__input"
                        style={
                          Object {
                            "display": "inline-block",
                            "fontSize": 14,
                          }
                        }
                      >
                        <input
                          aria-controls=""
                          data-test-subj="comboBoxSearchInput"
                          id="random_html_id"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="textbox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "2px",
                            }
                          }
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        />
                      </div>
                    </div>
                    <div
                      className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                    >
                      <button
                        aria-label="Open list of options"
                        className="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                        data-test-subj="comboBoxToggleListButton"
                        onClick={[Function]}
                        type="button"
                      >
                        <svg
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          height={16}
                          role="img"
                          style={null}
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13.069 5.157 8.384 9.768a.546.546 0 0 1-.768 0L2.93 5.158a.552.552 0 0 0-.771 0 .53.53 0 0 0 0 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 0 0 0-.76.552.552 0 0 0-.771 0Z"
                            fillRule="non-zero"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="euiSpacer euiSpacer--l"
          />
          <div
            className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
          >
            <div
              className="euiText euiText--medium"
            >
              <span
                className="panel-title"
              >
                Service map
              </span>
            </div>
            <div
              className="euiSpacer euiSpacer--m"
            />
            <fieldset
              className="euiButtonGroup euiButtonGroup--small euiButtonGroup--text"
              disabled={false}
            >
              <legend
                className="euiScreenReaderOnly"
              />
              <div
                className="euiButtonGroup__buttons"
              >
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small euiButtonGroupButton--fill euiButtonGroupButton-isSelected"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Duration"
                      title="Duration"
                    >
                      <input
                        checked={true}
                        className="euiScreenReaderOnly"
                        data-test-subj="latency"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Duration
                    </span>
                  </span>
                </label>
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Errors"
                      title="Errors"
                    >
                      <input
                        checked={false}
                        className="euiScreenReaderOnly"
                        data-test-subj="error_rate"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Errors
                    </span>
                  </span>
                </label>
                <label
                  className="euiButtonGroupButton euiButtonGroupButton--text euiButtonGroupButton--small"
                  disabled={false}
                  htmlFor="random_html_id"
                  onClick={[Function]}
                  style={
                    Object {
                      "minWidth": undefined,
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text euiButtonGroupButton__textShift"
                      data-text="Request Rate"
                      title="Request Rate"
                    >
                      <input
                        checked={false}
                        className="euiScreenReaderOnly"
                        data-test-subj="throughput"
                        disabled={false}
                        id="random_html_id"
                        name="random_html_id"
                        onChange={[Function]}
                        type="radio"
                      />
                      Request Rate
                    </span>
                  </span>
                </label>
              </div>
            </fieldset>
            <hr
              className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginMedium"
            />
            <div
              className="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  className="euiText euiText--medium"
                >
                  Focus on
                </div>
              </div>
              <div
                className="euiFlexItem"
              >
                <div
                  className="euiFormControlLayout"
                >
                  <div
                    className="euiFormControlLayout__childrenWrapper"
                  >
                    <input
                      className="euiFieldSearch"
                      onChange={[Function]}
                      onKeyUp={[Function]}
                      placeholder="Service name"
                      type="search"
                      value=""
                    />
                    <div
                      className="euiFormControlLayoutIcons"
                    >
                      <span
                        className="euiFormControlLayoutCustomIcon"
                      >
                        <svg
                          aria-hidden={true}
                          className="euiIcon euiIcon--medium euiFormControlLayoutCustomIcon__icon"
                          focusable="false"
                          height={16}
                          role="img"
                          style={null}
                          viewBox="0 0 16 16"
                          width={16}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="m11.271 11.978 3.872 3.873a.502.502 0 0 0 .708 0 .502.502 0 0 0 0-.708l-3.565-3.564c2.38-2.747 2.267-6.923-.342-9.532-2.73-2.73-7.17-2.73-9.898 0-2.728 2.729-2.728 7.17 0 9.9a6.955 6.955 0 0 0 4.949 2.05.5.5 0 0 0 0-1 5.96 5.96 0 0 1-4.242-1.757 6.01 6.01 0 0 1 0-8.486c2.337-2.34 6.143-2.34 8.484 0a6.01 6.01 0 0 1 0 8.486.5.5 0 0 0 .034.738Z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="euiSpacer euiSpacer--l"
            />
            <div
              style={
                Object {
                  "minHeight": 434,
                }
              }
            >
              Array [
                <div
                  className="euiSpacer euiSpacer--s"
                />,
                <div
                  className="euiEmptyPrompt"
                >
                  <h2
                    className="euiTitle euiTitle--medium"
                  >
                    No matches
                  </h2>
                  <span
                    className="euiTextColor euiTextColor--subdued"
                  >
                    <div
                      className="euiSpacer euiSpacer--m"
                    />
                    <div
                      className="euiText euiText--medium"
                    >
                      <div
                        className="euiText euiText--medium"
                      >
                        No data matches the selected filter. Clear the filter and/or increase the time range to see more results.
                      </div>
                    </div>
                  </span>
                </div>,
                <div
                  className="euiSpacer euiSpacer--s"
                />,
              ]
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
